### ISSUE TITLE
Document a better Roadmap

### ISSUE THREAD

**Issue by raoofm** [coreos/etcd-operator#1261]
A good roadmap helps us as users to better plan our work/releases. 

etcd does an amazing job at it and we know exactly when to look for a feature and plan our work accordingly.
https://github.com/coreos/etcd/blob/master/ROADMAP.md

I'm not sure if etcd-operator is in too early stage and switch priorities but some sort of a plan would be really helpful and also gives us an idea on when to look for the next stable upcoming releases with what new features.

**Comment by hongchaodeng** [comment]
What plan specifically do you want to know about or think that other users like to know?

**Comment by raoofm** [comment]
for instance, when is the stable release planned

**Comment by raoofm** [comment]
@hongchaodeng sorry for the noise, just checking in

**Comment by raoofm** [comment]
@hongchaodeng ping

**Comment by akauppi** [comment]
There's a [blog entry](https://sgotti.me/post/kubernetes-persistent-etcd/) that explains some of the concerns about etch-operator. I'm facing similar within the company - people saying operator is "not good (enough)". I don't really know what they mean (did not ask).

For this kind of clarity, a ROADMAP would be helpful.

**Comment by hongchaodeng** [comment]
Re. the blog, we are addressing the case right now with the author .

Please be patient. We rely on k8s feature which are not stable yet.

**Comment by raoofm** [comment]
@hongchaodeng @xiang90 
A roadmap doesn't have to be with a release date. Knowing about stable release date and what features to expect in say cycles of every 2 months helps to plan internally as well.

**Comment by hongchaodeng** [comment]
A major plan towards stable API: https://github.com/coreos/etcd-operator/issues/1626

I feel sorry to break users. But this is a price we need to pay to in order to make API minimal and stable.

After that, we will add features of PV/local PV for etcd data dir. No more features needed.

**Comment by hongchaodeng** [comment]
Grooming EtcdCluster API towards stable : https://github.com/coreos/etcd-operator/issues/1719

**Comment by akauppi** [comment]
Status: https://github.com/coreos/etcd-operator/blob/master/ROADMAP.md states "2017 Q1" and has last been edited in 2016.

I'm sorry to bring bad news, but if I were a newcomer to the project, that would be a turnoff for me. Despite the recent edits in the project - or maybe because of the contrast between those two things.

If there is a future plan for the etcd-operator, please update the ROADMAP. I need to know how much weight I can place on this, in a business setting.

**Comment by raoofm** [comment]
@akauppi I totally agree with you. I have been stressing the importance of this. For me the pull towards the projects etcd, kubernetes, vault etcd was the clear roadmap so that it helps in what to expect and to plan accordingly.

**Comment by raoofm** [comment]
@xiang90 I hope this isn't being abandoned?

**Comment by akauppi** [comment]
I'm back, checking etcd-operator in June 2018. 

Unfortunately, not much has been progressing. Checking through multiple issues today leaves me with the feeling that a fork might be a good thing. The etcd-operator is not the beef, etcd is. With the operator, people have strong use cases / needs, and trying to wrap them all into the same logic may be more difficult than having, say, 2-3 different operators, powerful in their own domains and focus.

For example: some operator could ditch v2 compatibility, for simplicity (there's an [issue for that](https://github.com/coreos/etcd-operator/issues/1315)). Another could do persistence with different focus than this one (there's an [issue for that](https://github.com/coreos/etcd-operator/issues/1323)). This one could remain a blessed, CoreOS variant.

Interestingly, if we compare this to the Linux ecosystem (kernel being etcd, operator being a distro), the people creating the kernel don't also create a distro. Those two things have different mindsets, and it may be best that different teams do them (+ we have multiple Linux distros, of course).

Obviously, I'm not blaming since this is open source. I just wish the whole ecosystem to flourish and the current pace is not sufficient for my needs.

**Comment by apuchitnis** [comment]
Following up on this thread. Does anyone know what additional work is required for the stable 1.0 release, and if RedHat/CoreOS are currently working on it? Or is this project effectively abandoned, and not being actively developed? I ask because I haven't seen a new release for the [last 6 months](https://github.com/coreos/etcd-operator/releases), and as this issue points out, there isn't a roadmap for future work.

**Comment by hexfusion** [comment]
@apuchitnis @akauppi work is continuing on this project and we would love your input/help.

**Comment by apuchitnis** [comment]
Awesome news @hexfusion. How could we help with the project?

**Comment by hexfusion** [comment]
@apuchitnis we are looking for help wherever you can offer. This includes answering questions/closing tickets, updating docs, helping with roadmap, adding features, improving testing. Eventually we would like to extend maintainership depending on your interest/time we have a lot of cycles to fill.

**Comment by raoofm** [comment]
> @apuchitnis we are looking for help wherever you can offer. This includes answering questions/closing tickets, updating docs, helping with roadmap, adding features, improving testing. Eventually we would like to extend maintainership depending on your interest/time we have a lot of cycles to fill.

@hexfusion I'll pitch in where I can
